DigitalLearningSolutions.Data.Tests/Services/RecommendedLearningServiceTests.cs

```diff
@@ -0,0 +1,295 @@
+    using DigitalLearningSolutions.Data.Services;
+    using FakeItEasy;
+    using FizzWare.NBuilder;
+    using FluentAssertions;
+    using NUnit.Framework;
+
+    public class RecommendedLearningServiceTests
+    {
+        private const int SelfAssessmentId = 1;
+        private const int CompetencyId = 2;
+        private const int LearningResourceReferenceId = 3;
+        private const int LearningHubResourceReferenceId = 4;
+        private const int DelegateId = 5;
+        private const int LearningLogId = 6;
+        private const string ResourceName = "Resource";
+        private const string ResourceDescription = "Description";
+        private const string ResourceCatalogue = "Catalogue";
+        private const string ResourceType = "link";
+        private const string ResourceLink = "www.test.com";
+
+        private ICompetencyLearningResourcesDataService competencyLearningResourcesDataService = null!;
+        private ILearningHubApiClient learningHubApiClient = null!;
+        private ILearningLogItemsDataService learningLogItemsDataService = null!;
+        private IRecommendedLearningService recommendedLearningService = null!;
+        private ISelfAssessmentDataService selfAssessmentDataService = null!;
+
+        [SetUp]
+        public void Setup()
+        {
+            competencyLearningResourcesDataService = A.Fake<ICompetencyLearningResourcesDataService>();
+            learningLogItemsDataService = A.Fake<ILearningLogItemsDataService>();
+            learningHubApiClient = A.Fake<ILearningHubApiClient>();
+            selfAssessmentDataService = A.Fake<ISelfAssessmentDataService>();
+
+            recommendedLearningService = new RecommendedLearningService(
+                selfAssessmentDataService,
+                competencyLearningResourcesDataService,
+                learningHubApiClient,
+                learningLogItemsDataService
+            );
+        }
+
+        [Test]
+        public async Task
+            GetRecommendedLearningForSelfAssessment_returns_correctly_populated_resource_when_learning_log_empty()
+        {
+            // Given
+            GivenResourceForSelfAssessmentIsReturnedByLearningHubApi();
+            A.CallTo(() => learningLogItemsDataService.GetLearningLogItems(DelegateId))
+                .Returns(new List<LearningLogItem>());
+
+            var expectedResource = new RecommendedResource
+            {
+                LearningResourceReferenceId = LearningResourceReferenceId,
+                LearningHubReferenceId = LearningHubResourceReferenceId,
+                ResourceName = ResourceName,
+                ResourceDescription = ResourceDescription,
+                ResourceType = ResourceType,
+                CatalogueName = ResourceCatalogue,
+                ResourceLink = ResourceLink,
+                IsInActionPlan = false,
+                IsCompleted = false,
+                LearningLogId = null,
+                RecommendationScore = 0,
+            };
+

```

Comment:
Since this is reporduced 4 times, maybe we could pull this out into a function like GetExpectedResource and pass in the values for IsInActionPlan and IsCompleted since those are the only values that change between the various tests I think?